experimental/DrinfeldHeckeAlgebras/src/DrinfeldHeckeAlgebra.jl

```diff
@@ -0,0 +1,669 @@
+
+Create the trivial Drinfeld-Hecke algebra over the ring ```R``` for the matrix group ```G```, i.e. the skew group ring $R[V]\#G$ 
+where $V$ is the vector space on which ```G``` acts
+
+# Examples
+```jldoctest
+julia> G = matrix_group(matrix(QQ, [-1 0;0 -1]))
+Matrix group of degree 2
+  over rational field
+
+julia> R, _ = polynomial_ring(QQ, ["x","y"])
+(Multivariate polynomial ring in 2 variables over QQ, QQMPolyRingElem[x, y])
+
+julia> drinfeld_hecke_algebra(G, R)
+Drinfeld-Hecke algebra
+   for Matrix group of degree 2 over QQ
+with generators
+   x1, x2, g1
+
+defined by Drinfeld-Hecke form over base ring
+   Multivariate polynomial ring in 2 variables over QQ
+with parameters 
+   x, y
+given by 0
+```
+"""
+drinfeld_hecke_algebra(G::MatrixGroup{T}, R::Ring=base_ring(G)) where {T <: FieldElem} = DrinfeldHeckeAlgebra(G, R)
+
+@doc raw"""
+    drinfeld_hecke_algebra(forms::Dict)
+
+Create the Drinfeld-Hecke algebra given by the dictionary ```forms``` which is expected to have keys of type 
+```MatrixGroupElem{T}``` and values of type ```MatElem{S}``` where ```T <: FieldElem, S <: RingElem```. If the types
+are not correct, an exception is thrown.
+
+# Examples
+```jldoctest
+julia> G = matrix_group(matrix(QQ, [-1 0;0 -1]))
+Matrix group of degree 2
+  over rational field
+
+julia> R, (x,y) = polynomial_ring(QQ, ["x","y"])
+(Multivariate polynomial ring in 2 variables over QQ, QQMPolyRingElem[x, y])
+
+julia> κ_1 = matrix(R, [0 x; -x 0])
+[ 0   x]
+[-x   0]
+
+julia> κ_g = matrix(R, [0 y; -y 0])
+[ 0   y]
+[-y   0]
+
+julia> forms = Dict(one(G) => κ_1, G[1] => κ_g)
+Dict{MatrixGroupElem{QQFieldElem, QQMatrix}, AbstractAlgebra.Generic.MatSpaceElem{QQMPolyRingElem}} with 2 entries:
+  [1 0; 0 1]   => [0 x; -x 0]
+  [-1 0; 0 -1] => [0 y; -y 0]
+
+julia> drinfeld_hecke_algebra(forms)
+Drinfeld-Hecke algebra
+   for Matrix group of degree 2 over QQ
+with generators
+   x1, x2, g1
+
+defined by Drinfeld-Hecke form over base ring
+   Multivariate polynomial ring in 2 variables over QQ
+with parameters 
+   x, y
+given by alternating bilinear forms
+   [1   0] => [ 0   x]
+   [0   1]    [-x   0]
+
+   [-1    0] => [ 0   y]
+   [ 0   -1]    [-y   0]
+```
+"""
+drinfeld_hecke_algebra(forms::Dict) = DrinfeldHeckeAlgebra(forms)
+
+#######################################
+# Drinfeld Hecke Algebra Element
+#######################################
+
+@doc raw"""
+    DrinfeldHeckeAlgebraElem{T <: FieldElem, S <: RingElem} <: NCRingElem
+
+Type for representing a Drinfeld-Hecke algebra element.
+"""
+mutable struct DrinfeldHeckeAlgebraElem{T <: FieldElem, S <: RingElem} <: NCRingElem
+  parent::DrinfeldHeckeAlgebra{T, S}
+  element::GroupAlgebraElem
+  
+  function DrinfeldHeckeAlgebraElem(A::DrinfeldHeckeAlgebra{T, S}, a::GroupAlgebraElem) where {T <: FieldElem, S <: RingElem}
+    if (a.parent != group_algebra(A))
+        throw(ArgumentError("Element does not belong to the underlying group algebra")) 
+    end
```

Comment:
More idiomatic would be
```suggestion
    @req a.parent != group_algebra(A) "Element does not belong to the underlying group algebra"
```